avaje-config/src/main/java/io/avaje/config/InitialLoader.java

```diff
@@ -188,12 +190,29 @@ private boolean loadTest() {
   }
 
   /**
-   * Load configuration defined by a <em>load.properties</em> entry in properties file.
+   * Recursively Load configuration defined by a <em>load.properties</em> entry in properties file.
    */
   private void loadViaIndirection() {
     String paths = loadContext.indirectLocation();
     if (paths != null) {
-      loadViaPaths(paths);
+      var stack = new ArrayDeque<String>();
+      splitAndAddPaths(stack, paths);
+      String path;
+      while ((path = stack.poll()) != null) {
+        loadWithExtensionCheck(loadContext.eval(path));
+        var newPath = loadContext.indirectLocation();
```

Comment:
newPath is nullable which will become paths and split on `null`.

Reply:
here `newpath` cannot be null, if the `load.properties` are not updated by a sub reference, the existing `load.properties` is returned. and you couldn't get that far if `load.properties` was null

Reply:
It is not about that it cannot be `null` now. It is that `loadContext.indirectLocation();` returns a `nullable` and can be nullable in the future by accident. Just think of it as returning an `Optional`.

Regardless you plugin checkerframework or whatever null checker and it will fail because `spitAndAddPaths` does not take a `@Nullable`.

Reply:
Basically just do this:

```java
var newPath = loadContext.indirectLocation();
if (newPath == null) throw new IllegalStateException("yada yada");
```